File: Python/Q-Bot.py
import json
import urllib.request

contactID = "ms458j"
to="sk093b"
Bot_from="ms458j"
botID = "ms458j@intl.att.com"
password = "Rocktheparty123"
botUrl = "http://bots.q.att.com:9015/QBotService/BotTalk?json=true"

# message = "Hi Suraj!"

message_file = open("message.txt","r+")
message = message_file.read()
print(message)
message_file.close()

body =     {
    "to":to,
    "from":botID,
    "password":password,
    "displayfromname":Bot_from,
    "html":message}

req = urllib.request.Request(botUrl)
req.add_header('Content-Type', 'application/json; charset=utf-8')
jsondata = json.dumps(body)
jsondataasbytes = jsondata.encode('utf-8')   # needs to be bytes
req.add_header('Content-Length', len(jsondataasbytes))
response = urllib.request.urlopen(req, jsondataasbytes)

File: Python/Q_bot.java
    
   public int send(String to, String botId, String botPassword, String friendlyBotName, String htmlMessage) throws IOException {
        URL botserviceURL = new URL("http://bots.q.att.com:9015/QBotService/BotTalk?json=true");
        HttpURLConnection botserviceConnection = (HttpURLConnection) botserviceURL.openConnection();
        botserviceConnection.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
        botserviceConnection.setRequestMethod("POST");
        botserviceConnection.setDoOutput(true);

        JsonObject message = new JsonObject();
        message.addProperty("to", to);
        message.addProperty("from", botId);
        message.addProperty("password", botPassword);
        message.addProperty("displayfromname", friendlyBotName);
        message.addProperty("html", htmlMessage);

        OutputStreamWriter writer = new OutputStreamWriter(botserviceConnection.getOutputStream());
        writer.write(message.toString());
        writer.flush();

        writer.close();
        botserviceConnection.disconnect();

        return botserviceConnection.getResponseCode();